packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/diagram/ViewToolFinder.java

```diff
@@ -0,0 +1,82 @@
+    }
+
+    @Override
+    public Optional<NodeTool> findNodeTool(IEditingContext editingContext, String diagramDescriptionId, String diagramElementDescriptionId, String toolId) {
+        return this.viewDiagramDescriptionSearchService.findById(editingContext, diagramDescriptionId)
+                .filter(viewDiagramDescription -> this.diagramIdProvider.getId(viewDiagramDescription).equals(diagramDescriptionId))
+                .flatMap(viewDiagramDescription -> this.getNodeToolByIdFromDiagramDescription(viewDiagramDescription, toolId))
+                .or(() -> this.viewDiagramDescriptionSearchService.findViewNodeDescriptionById(editingContext, diagramElementDescriptionId)
+                        .flatMap(viewNodeDescription -> this.getNodeToolByIdFromNodeDescription(viewNodeDescription, toolId)))
+                .or(() -> this.viewDiagramDescriptionSearchService.findViewEdgeDescriptionById(editingContext, diagramElementDescriptionId)
+                        .flatMap(viewEdgeDescription -> this.getNodeToolByIdFromEdgeDescription(viewEdgeDescription, toolId)));
+    }
+
+    @Override
+    public Optional<EdgeTool> findEdgeTool(IEditingContext editingContext, String diagramDescriptionId, String diagramElementDescriptionId, String toolId) {
+        return this.viewDiagramDescriptionSearchService.findViewNodeDescriptionById(editingContext, diagramElementDescriptionId)
+                .flatMap(viewNodeDescription -> this.getEdgeToolByIdFromNodeDescription(viewNodeDescription, toolId))
+                .or(() -> this.viewDiagramDescriptionSearchService.findViewEdgeDescriptionById(editingContext, diagramElementDescriptionId)
+                        .flatMap(viewNodeDescription -> this.getEdgeToolByIdFromEdgeDescription(viewNodeDescription, toolId)));
+    }
+
+    private Optional<NodeTool> getNodeToolByIdFromNodeDescription(NodeDescription viewNodeDescription, String toolId) {
+        return new ToolFinder().getNodeToolByIdFromNodeDescription(viewNodeDescription, toolId);
+    }
+
+    private Optional<NodeTool> getNodeToolByIdFromEdgeDescription(EdgeDescription viewEdgeDescription, String toolId) {
+        return new ToolFinder().getNodeToolByIdFromEdgeDescription(viewEdgeDescription, toolId);
+    }
+
+    private Optional<NodeTool> getNodeToolByIdFromDiagramDescription(org.eclipse.sirius.components.view.diagram.DiagramDescription diagramDescription, String toolId) {
+        return new  ToolFinder().getNodeToolByIdFromDiagramDescription(diagramDescription, toolId);
+    }
+
+    private Optional<EdgeTool> getEdgeToolByIdFromNodeDescription(NodeDescription viewNodeDescription, String toolId) {
+        return new ToolFinder().getEdgeToolByIdFromNodeDescription(viewNodeDescription, toolId);
+    }
+
+    private Optional<EdgeTool> getEdgeToolByIdFromEdgeDescription(EdgeDescription viewEdgeDescription, String toolId) {
+        return new ToolFinder().getEdgeToolByIdFromEdgeDescription(viewEdgeDescription, toolId);
+    }
```

Comment:
What is the point of those methods to just call another method with the same parameters?